src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -57,7 +57,15 @@ public PersonCard(Person person, int displayedIndex) {
         address.setText(person.getAddress().value);
         email.setText(person.getEmail().value);
         description.setText(person.getDescription().value);
-        nextOfKin.setText(person.getNextOfKin().value);
+
+        // Set text for nextOfKin label if value is present, otherwise make it invisible
+        person.getNextOfKin().ifPresentOrElse(
+                nok -> {
+                    nextOfKin.setText(nok.value);
+                    nextOfKin.setVisible(true);
+                }, () -> nextOfKin.setVisible(false)
```

Comment:
need to set not only visible but also nextOfKin.managed(false) as well. Try abstracting it out to a method,

hide(Node node) {
return () -> node.setVisible(false);
                   node.setManaged(false);
}

Also you would need setManaged(true) for the consumer then